modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerPipeliningTests.java

```diff
@@ -237,17 +245,15 @@ public void run() {
             final DefaultFullHttpResponse httpResponse = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK, buffer);
             httpResponse.headers().add(HttpHeaderNames.CONTENT_LENGTH, buffer.readableBytes());
 
-            final QueryStringDecoder decoder = new QueryStringDecoder(uri);
-
-            final int timeout =
-                uri.startsWith("/slow") && decoder.parameters().containsKey("sleep") ?
-                    Integer.valueOf(decoder.parameters().get("sleep").get(0)) : 0;
-            if (timeout > 0) {
+            final boolean slow = uri.matches("/slow/\\d+");
+            if (slow) {
                 try {
-                    Thread.sleep(timeout);
+                    Thread.sleep(scaledRandomIntBetween(500, 1000));
```

Comment:
The point of the test is that with pipelining disabled, a request that is slow to process should not prevent a response to a request that is fast to process. With pipelining enabled, we can assert that the responses come back in exactly the order that the requests were sent but with but with pipelining disabled, only that the responses to the slow requests come back last as long as the time to process the slow requests is slow enough. A sleep of 5 to 10 milliseconds is not slow enough, the processing of the slow and fast processes can race. A sleep of 500 to 1000 milliseconds gives a high degree of certainty that such racing is unlikely. Note that the total sleep time here is not ~ the sum of the sleeps across all the threads subject to sleeping, but instead more like ~ the max of the sleeps across all the threads subject to sleeping. This is because all the requests are processed concurrently.